scalableminds/webknossos
# Allow training a model on multiple annotations

### URL of deployed dev instance (used for testing):
- https://multiannotraining.webknossos.xyz
- I already created a few annotations (with differing configurations) and made the default user a super-admin
- details about the annotations:
	- https://multiannotraining.webknossos.xyz/annotations/66e2e04b010000df00b3f6fa
		- no-tracing segmentation layer: "segmentation"
	- https://multiannotraining.webknossos.xyz/annotations/66e2dfe6010000ab00b3f6f7
		- volume layer with fallback: "segmentation"
	- https://multiannotraining.webknossos.xyz/annotations/66e2e06d010000ad00b3f6ff
		- volume layer with fallback: "custom_segmentation"
	- https://multiannotraining.webknossos.xyz/annotations/66e2e08c010000ad00b3f702
		- volume layer with fallback: custom_segmentation
		- volume layer without fallback: custom_empty
			- finest mag is mag4
	- https://multiannotraining.webknossos.xyz/annotations/66e2e0d6010000ac00b3f706
		- no-tracing segmentation layer: "segmentation"
		- volume layer without fallback: custom_empty

### Steps to test:
- go to admin > ai models
- click on the "train new model" button
- paste this CSV
```
66e2e04b010000df00b3f6fa
https://multiannotraining.webknossos.xyz/annotations/66e2dfe6010000ab00b3f6f7
https://multiannotraining.webknossos.xyz/annotations/66e2e06d010000ad00b3f6ff
https://multiannotraining.webknossos.xyz/annotations/66e2e08c010000ad00b3f702
https://multiannotraining.webknossos.xyz/annotations/66e2e0d6010000ac00b3f706
```
- click "add"
- use the form
- submit it (it will fail because there is no worker configured on the dev instance). however, you can use the network tab to inspect the sent payload.

![image](https://github.com/user-attachments/assets/b9f25ce4-407c-428d-820c-d796b9a27daf)
![image](https://github.com/user-attachments/assets/8217b122-b208-458e-9e5d-6a824bcec831)
- also open an annotation and use the train-model dialog from there to ensure that it still works
- special care needs to be taken for the volume layer name, because we usually use the tracing id as the name and then have a human-readable name for the UI. however, the worker always wants the human readable one and never the tracing id.

### Issues:
- fixes #8005

------
